### PR TITLE
Compositors demos

### PR DESCRIPTION
**Pull request by Janix4000**

### CODE REVIEW

File: demo/mix.exs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  end
+
+  defp docs do
+    [
+      main: "readme",
+      extras: ["README.md", "LICENSE"],
+      formatters: ["html"],
+      source_ref: "v#{@version}",
+      nest_modules_by_prefix: [Membrane.VideoCompositor]
+    ]
```
Comment (DominikWolek) [inline demo/mix.exs:88]: I don't think we need docs in this project. A README would be nice, though.

File: demo/.formatter.exs [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Used by "mix format"
+[
+  inputs: ["{mix,.formatter}.exs", "{config,lib,test}/**/*.{ex,exs}"]
+]
```
Comment (DominikWolek) [inline demo/.formatter.exs:4]:
```suggestion
[
  inputs: ["{mix,.formatter}.exs", "{config,lib,test}/**/*.{ex,exs}"],
  import_deps: [:membrane_core]
]
```

File: pipeline/.formatter.exs [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# Used by "mix format"
+[
+  inputs: ["{mix,.formatter}.exs", "{config,lib,test}/**/*.{ex,exs}"]
+]
```
Comment (DominikWolek) [inline pipeline/.formatter.exs:4]:
```suggestion
[
  inputs: ["{mix,.formatter}.exs", "{config,lib,test}/**/*.{ex,exs}"],
  import_deps: [:membrane_core]
]
```

File: utility/.formatter.exs [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# Used by "mix format"
+[
+  inputs: ["{mix,.formatter}.exs", "{config,lib,test}/**/*.{ex,exs}"]
+]
```
Comment (DominikWolek) [inline utility/.formatter.exs:4]:
```suggestion
[
  inputs: ["{mix,.formatter}.exs", "{config,lib,test}/**/*.{ex,exs}"],
  import_deps: [:membrane_core]
]
```

File: demo/README.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+### Possible options:
+- `IMPL=` - implementation of video compositor 
+  - `ffmpeg`
+  - `nx`
+- `SINK=` - result destination
+  - `file` - create an output file with the corresponding video format
+  - `play` - play result in real-time, using SDL video player
+
+### Currently available demos:
+ - `lib/h264_pipeline_script.exs` - 60 seconds, 30 fps, full-hd (not support `SINK=play` right now)
```
Comment (DominikWolek) [inline demo/README.md:25]:
```suggestion
 - `lib/h264_pipeline_script.exs` - 60 seconds, 30 fps, full-hd (`SINK=play` is not supported right now)
```

File: demo/README.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+`IMPL=ffmpeg SINK=file  mix run lib/h264_pipeline_script.exs`
+
+### Possible options:
+- `IMPL=` - implementation of video compositor 
+  - `ffmpeg`
+  - `nx`
+- `SINK=` - result destination
+  - `file` - create an output file with the corresponding video format
+  - `play` - play result in real-time, using SDL video player
```
Comment (DominikWolek) [inline demo/README.md:22]: `System` module allows you to use argv easily, but I leave it up to you

File: demo/README.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+## Usage
+
+Run demos with the command line, for example:
+
+`IMPL=ffmpeg SINK=file  mix run lib/h264_pipeline_script.exs`
+
+### Possible options:
+- `IMPL=` - implementation of video compositor 
+  - `ffmpeg`
+  - `nx`
```
Comment (DominikWolek) [inline demo/README.md:19]: `opengl` is also available right now

File: demo/mix.exs [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+      deps: deps(),
+
+      # hex
+      description: "Demos for Video Compositor for Membrane Multimedia Framework",
+      package: package(),
+
+      # docs
+      name: "Membrane Video Compositor plugin",
+      source_url: @github_url,
+      homepage_url: "https://membraneframework.org"
```
Comment (DominikWolek) [inline demo/mix.exs:23]: Unecessary

File: demo/mix.exs [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+  # Run "mix help compile.app" to learn about applications.
+  def application do
+    [
+      extra_applications: [:logger]
+    ]
+  end
+
+  defp elixirc_paths(:test), do: ["lib"]
+  defp elixirc_paths(_env), do: ["lib"]
```
Comment (DominikWolek) [inline demo/mix.exs:35]: Unecessary

File: demo/mix.exs [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+
+  @version "0.1.0"
+  @github_url "https://github.com/membraneframework/membrane_video_compositor_plugin"
+
+  def project do
+    [
+      app: :membrane_video_compositor_plugin_demo,
+      version: @version,
+      elixir: "~> 1.13",
+      elixirc_paths: elixirc_paths(Mix.env()),
```
Comment (DominikWolek) [inline demo/mix.exs:12]:
```suggestion
      elixirc_paths: ["lib"],
```

File: demo/mix.exs [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+  defp package do
+    [
+      maintainers: ["Membrane Team"],
+      licenses: ["Apache-2.0"],
+      links: %{
+        "GitHub" => @github_url,
+        "Membrane Framework Homepage" => "https://membraneframework.org"
+      }
+    ]
+  end
```
Comment (DominikWolek) [inline demo/mix.exs:64]: Unecessary

File: pipeline/mix.exs [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+      deps: deps(),
+
+      # hex
+      description: "Testing pipelines for Video Compositor, for Membrane Multimedia Framework",
+      package: package(),
+
+      # docs
+      name: "Pipelines for Membrane Video Compositor plugin",
+      source_url: @github_url,
+      homepage_url: "https://membraneframework.org"
```
Comment (DominikWolek) [inline pipeline/mix.exs:23]: Unecessary

File: pipeline/mix.exs [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+
+  @version "0.1.0"
+  @github_url "https://github.com/membraneframework/membrane_video_compositor_plugin"
+
+  def project do
+    [
+      app: :membrane_video_compositor_plugin_pipeline,
+      version: @version,
+      elixir: "~> 1.13",
+      elixirc_paths: elixirc_paths(Mix.env()),
```
Comment (DominikWolek) [inline pipeline/mix.exs:12]:
```suggestion
      elixirc_paths: ["lib"],
```

File: pipeline/mix.exs [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+
+  # Run "mix help compile.app" to learn about applications.
+  def application do
+    [
+      extra_applications: [:logger]
+    ]
+  end
+
+  defp elixirc_paths(:test), do: ["lib"]
+  defp elixirc_paths(_env), do: ["lib"]
```
Comment (DominikWolek) [inline pipeline/mix.exs:35]: Unecessary

File: pipeline/mix.exs [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+  defp package do
+    [
+      maintainers: ["Membrane Team"],
+      licenses: ["Apache-2.0"],
+      links: %{
+        "GitHub" => @github_url,
+        "Membrane Framework Homepage" => "https://membraneframework.org"
+      }
+    ]
+  end
```
Comment (DominikWolek) [inline pipeline/mix.exs:58]: Unecessary

File: demo/README.md [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+ - `lib/h264_pipeline_script.exs` - 60 seconds, 30 fps, full-hd (not support `SINK=play` right now)
+ - `lib/h264_pipeline_script.exs` - 10 seconds, 1 fps, hd
+
+## Copyright and License
+
+Copyright 2022, [Software Mansion](https://swmansion.com/?utm_source=git&utm_medium=readme&utm_campaign=membrane_video_compositor_plugin)
+
+[![Software Mansion](https://logo.swmansion.com/logo?color=white&variant=desktop&width=200&tag=membrane-github)](https://swmansion.com/?utm_source=git&utm_medium=readme&utm_campaign=membrane_video_compositor_plugin)
+
+Licensed under the [Apache License, Version 2.0](LICENSE)
```
Comment (DominikWolek) [inline demo/README.md:34]: Link to licence in the root of the repo, or provide a licence file here in demo

File: pipeline/README.md [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+  [
+    {:pipeline, "~> 0.1.0"}
+  ]
+end
+```
+
+Documentation can be generated with [ExDoc](https://github.com/elixir-lang/ex_doc)
+and published on [HexDocs](https://hexdocs.pm). Once published, the docs can
+be found at <https://hexdocs.pm/pipeline>.
+
```
Comment (DominikWolek) [inline pipeline/README.md:21]: Unecessary - this probably never will be released

File: test/support/input_parser_h264.ex [hunk 18]
```diff
@@ -0,0 +1,32 @@
+defmodule Membrane.VideoCompositor.Test.Pipeline.H264.TwoInputsParser do
```
Comment (DominikWolek) [inline test/support/input_parser_h264.ex:1]: NIT: 
```suggestion
defmodule Membrane.VideoCompositor.Test.TwoInputsParser do
```

**Review by DominikWolek — CHANGES_REQUESTED** [review]

File: demo/README.md (hunk 6 above)
Comment (Janix4000) [reply demo/README.md:22]: I leave it as it is, right now

File: test/support/input_parser_h264.ex (hunk 18 above)
Comment (Janix4000) [reply test/support/input_parser_h264.ex:1]: But, it is for the testing pipeline, specifically for the H264 input.

**Review by DominikWolek — APPROVED** [review]

**Pull request merged**